pandas/core/indexes/multi.py

```diff
@@ -541,6 +555,13 @@ def from_product(cls, iterables, sortorder=None, names=None):
         elif is_iterator(iterables):
             iterables = list(iterables)
 
+        # Infer names from iterable if attribute is available
+        if names is None:
+            names = [idx.name if hasattr(idx, "name") else None for idx in iterables]
```

Comment:
[`getattr`](https://docs.python.org/3/library/functions.html#getattr) allows for a default value to be specified, so this can be simplified to:
```python
names = [getattr(idx, "name", None) for idx in iterables]
```

Reply:
Good catch, this is done now.